src/test/java/org/jabref/logic/citationkeypattern/CitationKeyGeneratorTest.java

```diff
@@ -1150,4 +1150,11 @@ void generateKeyWithLowercaseAuthorWithVonAndLastname() {
         entry.setField(StandardField.YEAR, "2021");
         assertEquals("Brekel2021", generateKey(entry, "[auth][year]"));
     }
+
+    @Test
+    void generateKeyCorrectKeyWithAndOthersAtTheEnd() {
+        BibEntry entry = createABibEntryAuthor("Alexander Artemenko and others");
+        entry.setField(StandardField.YEAR, "2019");
+        assertEquals("Artemenko2019", generateKey(entry, "[auth][year]"));
```

Comment:
This is OK here.

We need another test with `[authors]` and `[authorsAlpha]`. -- They are roughly explained at https://docs.jabref.org/setup/citationkeypatterns#special-field-markers.

- `[authors]` -> `Artemenko+`
- `[authorsAlpha]` -> `A+` (as far as I know. Can you setup a short Latex MWE to check?)

Reply:
Fixed in [`c8df469` (#9703)](https://github.com/JabRef/jabref/pull/9703/commits/c8df4694819060573fa5b2861fcd2ad89c22ce7a)

Reply:
[authshort]: The last name if one author is given; the first character of up to three authors' last names if more than one author is given. A plus character is added, if there are more than three authors.

Reply:
Fixed in [`b107f0a` (#9703)](https://github.com/JabRef/jabref/pull/9703/commits/b107f0a4e271d6ef7f3a88a8f93b6f557a5a052b)